src/main/java/org/springframework/integration/kafka/outbound/KafkaProducerMessageHandler.java

```diff
@@ -142,28 +144,15 @@ protected void handleMessageInternal(final Message<?> message) throws Exception
 				? this.messageKeyExpression.getValue(this.evaluationContext, message)
 				: message.getHeaders().get(KafkaHeaders.MESSAGE_KEY);
 
-		ListenableFuture<?> future;
+		Long timestamp = message.getHeaders().get(KafkaHeaders.TIMESTAMP, Long.class);
```

Comment:
Everything is good, @bijukunjummen, and we can accept PR, but then we need to start a new story for improvements I will request. So, let's address then here, too! 😄 

So, this `KafkaHeaders.TIMESTAMP` is good, but I really would like to see `timestampExpression` as well.
Absolutely similar what we have for `topic `, `partitionId` and `messageKey` here already.

The new expression option should be reflected in the `KafkaProducerMessageHandlerSpec` for Java DSL.

Of course, if you have some argument do not do that, I'd like to hear it.

Also, please, take a look into `README.adoc`. Maybe there is need to mention our new `timestamp` support as well.

And one more general concern from my side: commit message - https://github.com/spring-projects/spring-integration/blob/master/CONTRIBUTING.adoc#provide-a-link-to-the-jira-issue-in-the-associated-pull-request

We prefer:
* to have a GH issue number in the header of the message
* the header must be not longer than 50 symbols
* After header there must be a blank line
* The next line should be a link to the GH issue (JIRA) to have a cross-link traceability when you read commits history
* When we talk about GH issues, there are some rules to  have them automatically closed when we apply commit to `upstream`: https://help.github.com/articles/closing-issues-via-commit-messages/

Something like that. Just to make our life easier. 
Otherwise I forgot to polish your commit message and we don't know now for which story is a commit in master: https://github.com/spring-projects/spring-kafka/commit/8f43ad21335650a119b82061b08c7fab24ca34cc 😄